src/HDF5.jl

```diff
@@ -1378,26 +1376,41 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
+# Helper for reading compound types
+function read_row(io::IO, membertype, membersize)
+    row = Any[]
+    for (dtype, dsize) in zip(membertype, membersize)
+        if dtype === String
+            push!(row, strip(String(read(io, UInt8, dsize)),'\0'))
+        elseif dtype <: HDF5.FixedArray
+            push!(row, read(io, eltype(dtype), size(dtype)[1]))
```

Comment:
I still don't understand the `[1]` after `size(dtype)`. The definition:
```
size{T,D}(::Type{FixedArray{T,D}}) = D
```
ensures that you can only get back a single value when taking size of a `FixedArray`, so the extra indexing is unnecessary.